File: QuotaPetsShared/Sources/QuotaPetsShared/Usage/Codex/CodexResetTimestamp.swift
import Foundation

/// Decodes Codex's `reset_at`.
///
/// Ported concept: Orca `codex-rate-limit-window-mapper.ts:14-16`, whose comment is
/// explicit — *"Codex returns resetsAt as Unix seconds, not milliseconds."*
///
/// Unlike `ClaudeResetTimestamp` there is **no magnitude heuristic**: the unit is known.
/// Applying Claude's heuristic here would silently mis-scale any value above 1e10, and
/// applying this rule to Claude would put millisecond timestamps in the year 56 000.
/// The duplication between the two types is intentional (research §3.5).
public enum CodexResetTimestamp {
    public static func date(fromUnixSeconds value: Double?) -> Date? {
        guard let value, value.isFinite, value > 0 else { return nil }
        return Date(timeIntervalSince1970: value)
    }
}

File: src/usage/codexResetTimestamp.ts
/**
 * Decodes Codex's `reset_at`.
 *
 * Ported from Orca `codex-rate-limit-window-mapper.ts:14-16`, whose comment is explicit:
 * *"Codex returns resetsAt as Unix seconds, not milliseconds."*
 *
 * Unlike the Claude decoder there is **no magnitude heuristic**: the unit is known.
 * Applying Claude's rule here would silently mis-scale any value above 1e10, and applying
 * this rule to Claude would put millisecond timestamps in the year 56 000. The duplication
 * between the two is intentional.
 */
export function codexResetAt(value: unknown): number | null {
  if (typeof value !== "number") return null;
  if (!Number.isFinite(value) || value <= 0) return null;
  return value * 1000;
}
